bin/create-test-release.sh

```diff
@@ -0,0 +1,272 @@
+
+	# Update the 'Version' tag in the plugin headers to the current VERSION-PREFFIX
+	zip_file=$(echo $output | sed -n "s/.*Skipping upload of asset \(.*\) to GH release.*/\1/p")
+	# [ $? -eq 0 ] || exit 1
+
+    cd "$(dirname "$zip_file")/.."
+
+	echo ''
+	echo "> sed -i '' \"s/^ \* Version: .*$/ * Version: $VERSION-$SUFFIX/\" $GITHUB_PROJECT/$GITHUB_PROJECT.php"
+    sed -i '' "s/^ \* Version: .*$/ * Version: $VERSION-$SUFFIX/" $GITHUB_PROJECT/$GITHUB_PROJECT.php
+
+    echo ''
+	echo "> zip --update '$GITHUB_PROJECT/$GITHUB_PROJECT.zip' '$GITHUB_PROJECT/$GITHUB_PROJECT.php"
+    zip --delete "$GITHUB_PROJECT/$GITHUB_PROJECT.zip" "$GITHUB_PROJECT/$GITHUB_PROJECT.php" >/dev/null
+	zip --update "$GITHUB_PROJECT/$GITHUB_PROJECT.zip" "$GITHUB_PROJECT/$GITHUB_PROJECT.php" >/dev/null
+
+    # Copy the plugin zip file to the working dir, in case manual upload to GitHub is necessary
+	cd - >/dev/null
+    cp $zip_file .
+}
+
+create_github_release() {
+	tag="$VERSION-$SUFFIX"
+	name="Version $tag. Not for production."
+	body='This version is for internal testing. It should NOT be used for production.'
+
+	echo ''
+	echo "Creating test release '$tag'..."
+
+	response_output=$(mktemp)
+
+	http_code=$(curl --request POST \
+		--header "Authorization: token $GITHUB_TOKEN" \
+		--header 'Content-Type: application/json' \
+		--data "{\"tag_name\":\"$tag\",\"target_commitish\":\"release/$VERSION\",\"name\":\"$name\",\"body\":\"$body\",\"prerelease\":true}" \
+		--silent \
+		--output "$response_output" \
+		--write-out "%{http_code}" \
+		https://api.github.com/repos/$GITHUB_ACCOUNT/$GITHUB_PROJECT/releases)
+
+	response=$(cat "$response_output")
+	rm "$response_output"
+
+	if [[ ${http_code} -lt 200 || ${http_code} -gt 299 ]]; then
+		echo "Could not create release:"
+		echo "$response"
+		exit 1
+	fi
+
+	echo ''
+	echo "Uploading plugin zip package to the GitHub Release..."
+
+	upload_url=$(echo $response | grep 'upload_url' | sed 's|.*"upload_url": "\([^{]*\).*|\1|')
+	if [[ -z $upload_url ]]; then
+		echo "No upload URL found in response:"
+		echo "$response"
+		exit 1
+	fi
+
+	response_output=$(mktemp)
+
+	http_code=$(curl --request POST \
+		--header "Authorization: token $GITHUB_TOKEN" \
+		--header 'Content-Type: application/zip' \
+		--data-binary @$GITHUB_PROJECT.zip \
+		--silent \
+		--output "$response_output" \
+		--write-out "%{http_code}" \
+		"$upload_url?name=$GITHUB_PROJECT.zip")
+
+	response=$(cat "$response_output")
+	rm "$response_output"
+
+	if [[ ${http_code} -lt 200 || ${http_code} -gt 299 ]]; then
+		echo "Could not upload plugin zip to GitHub release:"
+		echo "$response"
+		exit 1
+	fi
+}
+
+# -----
+
+# Check for 'git', 'svn, 'php' and 'Woorelease'
+check_prerequisites
+
+# Check if the 'release/VERSION' branch exists, and the next test SUFFIX to use (ex: 5.3.0-test3)
+check_github_status
+
+# Print the actions to perform
+print_summary
+
+echo ''
+read -p "Do you want to continue? (yes/no) " -n 1 -r
```

Comment:
The `yes/no` should be `y/N`; as soon as you press <kbd>y</kbd> it continues, any other letter and it stops.

Reply:
Changed